### PR TITLE
Aida 303

### PR DESCRIPTION
**Pull request by jakevb**
Run main of OntologyGeneration with the path to the ontologies you wish to use as arguments.  README located in src/main/java/com/ncc/aif/OntToJavaGen

### CODE REVIEW

File: src/main/java/com/ncc/aif/OntToJavaGen/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Running OntologyGeneration
+
+This class will create Java classes based on the ontology or ontologies used as arguments when running Main.
```
Comment (nextcen-dgemoets) [inline src/main/java/com/ncc/aif/OntToJavaGen/README.md]: Out of curiosity, why is this in a separate README, and in a source code directory?  It seems that this could just be incorporated into a section in the main README.md, maybe after "Running the validator."

Also, I think the README should advise that after generating this file, developers will need to write an ontology-specific subclass of OntologyMapper that uses the constants in the generated file.  They also might want to write ontology-specific tests (e.g., another version of ExamplesAndValidationTests.

File: src/main/java/com/ncc/aif/OntToJavaGen/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+The class will appear in `src/main/java/com/ncc/aif` and the name of the class created will be based on the
+name of the ontology used as the program argument.
+
+The class will just contain public static Resource variables based on the names of the classes specified in the given
+Ontologies. If more than one ontology is passed in as an argument, the number of classes created will be equal to the number of
+Ontologies used.
+
+To specify the program arguments in your IDE, navigate to the Edit Configurables for the OntologyGeneration class. Then add the path
+to the Ontology as the argument.
```
Comment (nextcen-dgemoets) [inline src/main/java/com/ncc/aif/OntToJavaGen/README.md]: For me (on Windows and Linux), it's Edit Configuration, not Configurables.

File: src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        }
+
+        String ending = "}";
+        owgMapping.add(ending);
+
+        return owgMapping;
+    }
+
+    private static List<String> owgMapperHeader () {
+
```
Comment (nextcen-dgemoets) [inline src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java]: I think it would be nice to have a brief, generated file and/or class comment that:
- Mentions that this is a generated file (and "DO NOT EDIT" warning);
- Includes the date of generation;
- Gives a one-liner on what the file is and/or refers to the README.
- If it's available programmatically, the ontology’s comment, label, version info (rdfs:comment, rdfs:label, owl:versionInfo).

File: pom.xml [hunk 4]
```diff
@@ -215,7 +215,6 @@
                         <artifactId>junit-jupiter-engine</artifactId>
                         <version>5.2.0</version>
                     </dependency>
-
```
Comment (nextcen-dgemoets) [inline pom.xml]: It looks like there's just an accidental whitespace change here.  If pom.xml isn't really modified, it shouldn't be part of the PR.

**Comment by jakevb** [comment]
I apologize, but I seemed that I forgot to add the generated ontology constants class as well as the current Ontology Working Group Ontologies to the git project.  This might have caused some confusion, as the current purpose is not explicitly for users to create the Ontology resources from their own personal ontologies (although they can if they wish too), but for us to update the code base when updates or new ontologies are added to the project.  It will also be less confusing for the users as they will not need to run mvn install twice or other counter-intuitive measures to leverage use of the Ontology resources.  As such, I am not sure about adding the use of OntologyGenerator as part of the top level README.

File: src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java (hunk 3 above)
Comment (nextcen-dgemoets) [reply src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java]: For the record:  the comment, label, and version info weren't easily obtainable programmatically.  Resolving this comment.

File: src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java [hunk 5]
```diff
@@ -33,11 +33,7 @@ private OntologyGeneration() {
         this.classesToGenerate = new ArrayList();
```
Comment (nextcen-dgemoets) [inline src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java]: What is classesToGenerate used for?  It's declared, initialized, and URIs are added to it, but then it's never used that I can tell.

File: src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+/**
+ * Generates java resources  from an ontology
+ */
+public class OntologyGeneration {
```
[the remark this answers is not in the document]
Comment (nextcen-dgemoets) [reply src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java]: Skip the @author tag since it doesn't look like we use it much on this project.  But the class comment should provide more information/instructions or refer to the README.

File: src/main/java/com/ncc/aif/OntToJavaGen/README.md (hunk 1 above)
Comment (jakevb) [reply src/main/java/com/ncc/aif/OntToJavaGen/README.md]: This class is primarily used for Next Century to update the ontology and add new ontologies whenever changes are made.  While this class is available to use by other users, that is not it's primary purpose.  As such, A segmented README was created in the source code directory for just this class with a link in the top-level directory referencing this README for users who wish to use our Java Generation class.

File: src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java (hunk 5 above)
Comment (jakevb) [reply src/main/java/com/ncc/aif/ont2javagen/OntologyGeneration.java]: Code was vestigial and has been removed.

File: src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (jakevb) [reply src/main/java/com/ncc/aif/OntToJavaGen/OntologyGeneration.java]: Reference to README was added.

File: README.md [hunk 7]
```diff
@@ -75,6 +75,11 @@ ontology to validate against.  Beware that validating large KBs can take a long
 a sample of a validator param file in `sample_params/validate.common_corpus.single.params`
 * To run the validator using the Kotlin version, run `target/appassembler/bin/validateAIF`.
 
+# Running the Ontology Resource Generator
+
+To generate the resource variables from a particular ontology file, please refer to 
+the README located at `src/test/java/com/ncc/aif/ont2javagen/README.md`.
+
```
Comment (nextcen-dgemoets) [inline README.md]: The file is in src/main/java, not src/test/java.  But I actually kind of prefer it in src/test/java, since using this generator on a new ontology will generally involve writing a new (or seriously updating) ExamplesAndValidationTests.java.  But I'm okay with either.

File: src/main/java/com/ncc/aif/OntToJavaGen/README.md (hunk 1 above)
Comment (nextcen-dgemoets) [reply src/main/java/com/ncc/aif/OntToJavaGen/README.md]: I think the README should advise developers that they'll need to write ontology-specific tests (e.g., another version of ExamplesAndValidationTests), since that's part of the process and presumably we'll do that for the program ontology.  But I'm not hung up on it-- I'm resolving the comment.

File: README.md (hunk 7 above)
Comment (craigwarsaw) [reply README.md]: It should be in src/main/java.  It is not a unit test.  Also, to Darren's comment, the ExamplesAndValidationTests.java is not going to be the only use of this (although it might be OUR only use of it).

File: src/main/resources/com/ncc/aif/EntityOntology [hunk 8]
```diff
@@ -0,0 +1,1539 @@
...
+@prefix schema: <http://schema.org/> .
+@prefix olia:  <http://purl.org/olia/system.owl#> .
+@prefix aidaDomainCommon: <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/AidaDomainOntologiesCommon#> .
+@prefix evt:   <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/EventsWithFringe#> .
+@prefix rdf:   <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix owl:   <http://www.w3.org/2002/07/owl#> .
+@prefix rel:   <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/EntityRelations#> .
+@prefix aida:  <https://tac.nist.gov/tracks/SM-KBP/2018/ontologies/InterchangeOntology#> .
+@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .
+
```
[the remark this answers is not in the document]
Comment (craigwarsaw) [reply src/main/resources/com/ncc/aif/EntityOntology]: After consideration, I think we leave the program ontologies named as is.  They will be the long term ontologies - the LDC ontology is the short term one used for evaluation purposes.

[the remark this answers is not in the document]
Comment (nextcen-dgemoets) [reply src/main/resources/com/ncc/aif/EntityOntology]: Okay, but it sounds like you still want them moved to a subdirectory off of resources/com/ncc/aif, right?

[the remark this answers is not in the document]
Comment (craigwarsaw) [reply src/main/resources/com/ncc/aif/EntityOntology]: Yes

File: src/test/java/com/ncc/aif/ExamplesAndValidationTest.java [hunk 9]
```diff
@@ -1007,7 +975,7 @@ void createEntityWithDiskBaseModelAndWriteOut() {
             final Resource system = makeSystemWithURI(model, getTestSystemUri());
 
             final Resource entity = makeEntity(model, putinDocumentEntityUri, system);
-            final Resource entityIsAPerson = markType(model, getAssertionUri(), entity, SeedlingOntologyMapper.PERSON,
+            final Resource entityIsAPerson = markType(model, getAssertionUri(), entity, SeedlingOntology.Person,
                     system, 0.5);
 //        final Resource entityIsAPoliticalEntity = markType(model, getAssertionUri(), entity,
 //                SeedlingOntologyMapper.GPE, system, 0.2);
```
[the remark this answers is not in the document]
Comment (jakevb) [reply src/test/java/com/ncc/aif/ExamplesAndValidationTest.java]: The comment was updated to refer to the SeedlingOntology constants.
